lib/openstack_query/managers/server_manager.py

```diff
@@ -26,168 +27,114 @@ class ServerManager(QueryManager):
+            output_details=QueryOutputDetails.from_kwargs(
+                prop_cls=ServerProperties, **kwargs
             ),
-            output_details=output_details,
         )
 
-    def search_servers_last_updated_after_relative_to_now(
-        self,
-        output_details: QueryOutputDetails,
-        days: int,
-        hours: int = 0,
-        minutes: int = 0,
-        seconds: int = 0,
+    def search_by_property(
+        self, search_mode: bool, property_to_search_by: str, values: List[str], **kwargs
     ) -> QueryReturn:
         """
-        method that returns a list of all servers which were last updated after a time relative to now.
-        Uses UTC timezone
-        :param output_details: A dataclass containing config info on how results should be returned
-        :param days: number of days since current time
-        :param hours: number of hours since current time
-        :param minutes: number of minutes since current time
-        :param seconds: number of seconds since current time
+        method that builds and runs a query to find Openstack servers with a selected property
+        matching, or not matching given value(s)
+        :param search_mode: A boolean representing what to return from query, if True - use the preset ANY_IN/EQUAL_TO,
+        if False - use the preset NOT_ANY_IN/NOT_EQUAL_TO
+        :param property_to_search_by: A string representing a datetime property Enum that the preset will be used on
+        :param values: A list of string values to compare server property against
+        :param kwargs: A set of optional kwargs to pass to the query
         """
-        return self._build_and_run_query(
-            preset_details=QueryPresetDetails(
-                preset=QueryPresetsDateTime.YOUNGER_THAN,
-                prop=ServerProperties.SERVER_LAST_UPDATED_DATE,
-                args={
-                    "days": days,
-                    "hours": hours,
-                    "minutes": minutes,
-                    "seconds": float(seconds),
-                },
-            ),
-            output_details=output_details,
+        args = {"values": values}
+        preset = (
+            QueryPresetsString.ANY_IN if search_mode else QueryPresetsString.NOT_ANY_IN
         )
 
-    def search_servers_name_in(
-        self, output_details: QueryOutputDetails, names: List[str]
-    ) -> QueryReturn:
-        """
-        method that returns a list of all servers which have a name in a given list
-        :param output_details: A dataclass containing config info on how results should be returned
-        :param names: a list of server names to check against
-        """
+        # If values contains only one value - use EQUAL_TO/NOT_EQUAL_TO as the preset instead to speed up query
+        if len(values) == 1:
+            res = {
+                QueryPresetsString.ANY_IN: QueryPresetsGeneric.EQUAL_TO,
+                QueryPresetsString.NOT_ANY_IN: QueryPresetsGeneric.NOT_EQUAL_TO,
```

Comment:
I can't think of a way, but it would be nice to get whatever base query runner (we abstract this to in the future) to do the fixup for all it's delegated classes, since it's an easy win

Reply:
I don't understand what you mean?